src/gui/tray/ActivityItem.qml

```diff
@@ -0,0 +1,288 @@
+import QtQml 2.12
+import QtQuick 2.9
+import QtQuick.Controls 2.2
+import QtQuick.Layouts 1.2
+import Style 1.0
+
+MouseArea {
+    id: activityMouseArea
+    enabled: (path !== "" || link !== "")
+    width: Style.trayWindowWidth
+    height: Style.trayWindowHeaderHeight
+    Layout.alignment: Qt.AlignLeft
+    Layout.margins: 2
+    Layout.fillWidth: true
+    Layout.fillHeight: true
+    hoverEnabled: true
+    onClicked: activityModel.triggerDefaultAction(model.index)
+    
+    Rectangle {
+        Layout.fillWidth: true
+        Layout.fillHeight: true
+        width: Style.trayWindowWidth
+        height: Style.trayWindowHeaderHeight
+        color: (parent.containsMouse ? Style.lightHover : "transparent")
+        
+        RowLayout {
+            id: activityItem
+            Layout.fillWidth: true
+            Layout.fillHeight: true
+            
+            readonly property variant links: model.links
+            
+            readonly property int itemIndex: model.index
+            
+            width: activityMouseArea.width
```

Comment:
if you decide to go with the anchors to set the size, you should not also use directory `width` and `height`